src/main/java/com/microsoft/graph/exceptions/ServiceException.java

```diff
@@ -0,0 +1,165 @@
+package com.microsoft.graph.exceptions;
+
+import com.microsoft.kiota.ApiException;
+import com.microsoft.kiota.serialization.AdditionalDataHolder;
+import com.microsoft.kiota.serialization.Parsable;
+import com.microsoft.kiota.serialization.ParseNode;
+import com.microsoft.kiota.serialization.SerializationWriter;
+import okhttp3.Headers;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.annotation.Nonnull;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+/**
+ * Graph Service Exception
+ */
+public class ServiceException extends ApiException implements Parsable, AdditionalDataHolder {
+
+    /** The response headers received. */
+    private final transient Headers responseHeaders;
+    /** The response body received. */
+    private String rawResponseBody;
```

Comment:
can be `final`

Reply:
the getFieldDeserializers needs to be able to set the rawResponseBody so we can't set it to as that would remove our ability to set the field.
```Java
deserializers.put("rawResponseBody", n -> currentObj.rawResponseBody = n.getStringValue());
```
Making it final would cause the error: "Cannot assign a value to final variable 'rawResponseBody'"